File: lib/prompts/templates/warmup.ts
import type { Blueprint, GenerateRequest, SectionSpec } from '../../types'

export function buildPrompt(
  spec: SectionSpec,
  blueprint: Blueprint,
  req: GenerateRequest
): string {
  const { lesson_metadata } = req
  return `Create a warm-up activity for this study guide.

## Blueprint Context
- Lesson Title: ${blueprint.title}
- Core Concept: ${blueprint.core_concept}
- Essential Question: ${blueprint.essential_question}
- Subject: ${lesson_metadata.subject}
- Grade Level: ${lesson_metadata.grade_level}
- Market: ${lesson_metadata.market}

## Instructions
Design a short warm-up activity (5–10 minutes) that:
1. Activates prior knowledge relevant to this lesson
2. Is appropriate for Grade ${lesson_metadata.grade_level} ${lesson_metadata.subject}
3. Connects to the core concept: ${blueprint.core_concept}
4. Uses concrete, locally relevant examples for ${lesson_metadata.market} learners
5. Instructions should be numbered steps (2–3 steps)

## Output Schema
{
  "activity_title": "string (short, descriptive title)",
  "instructions": "string (2–3 numbered steps, e.g. '1. ... 2. ... 3. ...')",
  "purpose": "string (1 sentence explaining what prior skill or knowledge this activates)"
}

Section spec: ${JSON.stringify(spec)}`
}

File: backend/study-guide-agent/app/prompts/templates/warmup.py
from __future__ import annotations

from app.types import Blueprint, GenerateRequest


def build_prompt(spec, blueprint: Blueprint, request: GenerateRequest) -> str:
    del spec, request

    warmup_schema = """{
    "title": "Warm-Up",
    "activity_type": "string",
    "purpose": "string",
    "student_instructions": [
        "string"
    ],
    "teacher_tip": "string",
    "estimated_minutes": 5
}"""

    prompt_lines = [
        "Create the warm-up section for a K-12 study guide.",
        f"- Lesson title: {blueprint.title}",
        f"- Essential question: {blueprint.essential_question}",
        f"- Introduction hook: {blueprint.introduction_hook}",
        f"- Core concept: {blueprint.core_concept}",
        "Requirements:",
        "- Design one short activation activity that can be completed in about 5 minutes.",
        "- Use the hook and essential question to activate prior knowledge.",
        "- Include a clear purpose statement explaining why the warm-up matters for the lesson.",
        "- Provide student_instructions as a short ordered sequence the learner can follow.",
        "- Keep the activity lightweight, classroom-ready, and aligned to the lesson focus.",
        "Expected JSON schema:",
        warmup_schema,
        "Return only JSON.",
    ]

    return "\n".join(prompt_lines)
